srw/srw.cpp

```diff
@@ -506,6 +569,24 @@ class srw {
 			return err;
 		}
 
+		void complete_job(std::string id, sph_wrapper sph)
+		{
+			char sph_str[DNET_DUMP_NUM * 2 + 1];
+			dnet_dump_id_len_raw(sph.sph.src.id, DNET_DUMP_NUM, sph_str);
+			sph_str[2 * DNET_DUMP_NUM] = '\0';
```

Comment:
Please move it into error path.
Also, you can use `dnet_dump_id_str()`, it is thread-safe